pages/products/index.tsx

```diff
@@ -30,21 +30,19 @@ type ProductListingProps = Readonly<{
 }>
 
 const createQueryStringUrl = (params: Record<string, string | string[] | undefined>) =>
-  Object.keys(params).length > 0 ?
-    Object.entries(params).reduce(
-      (prev, next, index) => {
-        const [paramKey, paramValue] = next;
+  Object.keys(params).length > 0 ? '?'+
+    Object.entries(params).map(
+      (param) => {
+        const [paramKey, paramValue] = param;
```

Comment:
You can deconstruct the tuple right in the parameter definition like so `([paramKey, paramValue]) => ...` 😉